### PR TITLE
Add subscriptions for artifact jobs

### PR DESCRIPTION
**Pull request by e-edling-ericsson**
### Description of the Change

- Add subscriptions inside the Artifact flow tests.
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: 
Erik Edling erik.edling@ericsson.com

### CODE REVIEW

File: src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java [hunk 1]
```diff
@@ -39,4 +42,22 @@ public void initRemRemConfig() {
...
     }
+
+    /**
+     *
+     * This method retrieves the EI front end properties from the system properties.
+     *
+     * @return
+     */
+    public void initEIFrontend() {
+        eiFrontendBaseUrl = System.getProperty("EI_FRONTEND_BASE_URL");
```
Comment (tobiasake) [inline src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java]: Is this property going to be changed later to the real property name? e.g. "ei.frontend.base.url" or what it is named.

File: src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java [hunk 2]
```diff
@@ -39,4 +42,22 @@ public void initRemRemConfig() {
...
+
+        assertNotNull(eiFrontendBaseUrl);
+    }
+
+    public void initEIBackend() {
+        eiBackendBaseUrl = System.getProperty("EI_BACKEND_BASE_URL");
+
```
Comment (tobiasake) [inline src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java]: Same comment here.

File: src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java (hunk 1 above)
Comment (e-edling-ericsson) [reply src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java]: It should be changed to that in this PR :)

File: src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java (hunk 2 above)
Comment (e-edling-ericsson) [reply src/systemtest/java/com/ericsson/ei/systemtest/utils/Config.java]: It should be changed to that in this PR :)

File: src/systemtest/java/com/ericsson/ei/systemtest/utils/StepsUtils.java [hunk 3]
```diff
@@ -63,4 +71,96 @@ public static void deleteJenkinsJobs(ArrayList<String> jenkinsJobNames) throws E
             }
         }
     }
+
+    /**
+     * Creates a subscription that later can be used to send to Eiffel intelligence.
+     *
+     * @param subscriptionName
+     * @param nameOfTriggeredJob
```
Comment (a-breid-ericsson) [inline src/systemtest/java/com/ericsson/ei/systemtest/utils/StepsUtils.java]: Is the job already triggered?

File: src/systemtest/resources/features/ArtifactFlow.feature [hunk 4]
```diff
@@ -4,13 +4,53 @@ Feature: Artifact system test
...
     And a jenkins job '"CLMJob"' from '"src/systemtest/resources/JenkinsShellScripts/CLMScript.txt"' is created
-    And a jenkins job '"ArtP"' from '"src/systemtest/resources/JenkinsShellScripts/ArtPScript.txt"' is created
-    When next story happens
-    Then all is good
+    And a jenkins job '"ArtPJob"' from '"src/systemtest/resources/JenkinsShellScripts/ArtPScript.txt"' is created
+    And a jenkins job '"FlowCompleteJob"' from '"src/systemtest/resources/JenkinsShellScripts/FlowCompleteScript.txt"' is created
+    Then we continue with the next step
+
+    #####Add CLMSubscription to EI#####
+    Given subscription "CLMSubscription" is created which will trigger "CLMJob"
```
Comment (a-breid-ericsson) [inline src/systemtest/resources/features/ArtifactFlow.feature]: You create the subscription data, I was confused thinking that you sent it to EI at this step.

File: src/systemtest/resources/features/ArtifactFlow.feature [hunk 5]
```diff
@@ -4,13 +4,53 @@ Feature: Artifact system test
   @ArtifactFlowScenario
   Scenario: Artifact Flow Scenario
```
Comment (a-breid-ericsson) [inline src/systemtest/resources/features/ArtifactFlow.feature]: I would want to have a short summary what the scenario does. You may add it in multi line comment under the scenario: tag, no " or * or anything needed. it till be marked as just white text in a proper terminal when executing the test.

**Pull request merged**